rm/rm.h

```diff
@@ -764,6 +762,21 @@ struct m0_rm_remote {
 	 * resource manager service.
 	 */
 	struct m0_cookie        rem_cookie;
+	/**
+	 * Reference counter for the loans taken by this remote debtor.
+	 * When the last loan is settled (paid back) and there are no more
+	 * pending incoming requests (see rem_incomings), rm_remote debtor
+	 * instance can be freed.
+	 * Note: this should not be used for creditors whose life-cycle
+	 *       does not depend on the number of credited loans.
+	 */
+	struct m0_ref           rem_loans;
+	/**
+	 * Reference counter for pending rm_incoming requests.
+	 * rm_remote instance can be freed only if there are no pending
+	 * incoming requests on it (see rem_loans also).
+	 */
+	struct m0_ref           rem_incomings;
```

Comment:
Why do we need 2 separate reference counters? Wouldn't a single counter be sufficient?

Note that documentation is contradictory: first, it claims that a loan can be freed when there are no pending incoming requests *and* no unsettled loans, then that only "no pending incoming requests" condition is sufficient.

Reply:
Unfortunately, not. All loans can be taken from the debtor, but there might be still some pending requests from it (he still might want the loans). In this case we don't want to free the debtor.

The 2nd comment states: "rm_remote instance can be freed only if there are no pending incoming requests on it". It does not claim that it should be a sufficient condition to free it, but only that it is a necessary condition.

Reply:
I still fail to understand how one reference counter is not enough.

What scenario would not work, if you replace all `m0_ref_get(&r->rem_{loans,incoming})` with `m0_ref_get(&r->rem_ref)` and all `m0_ref_put(&r->rem_{loans,incoming}, &func)` with `m0_ref_put(&r->rem_ref, &func)`?

Reply:
Fixed to one ref-counter, see https://github.com/Seagate/cortx-motr/pull/304/commits/990be42f16b64ce3a5b13ace4b310ca2fab1c0d4. Thanks for the hint!